### ISSUE TITLE
Disallow short struct syntax for functions with compulsory arguments and a trailing struct

### ISSUE THREAD

**Issue by hungrybluedev** [vlang/v#11209]
V version: V 0.2.2 fe5e3c4. 8521e22

**TL;DR** Don't allow mixing commas and newlines for separation of parameters and short form struct field declarations in function calls (for example).

The following snippet compiles and runs:
```
struct FooConfig {
	bar int = 2
	baz int = 2
}

fn foo(mandatory int, config FooConfig) {
	println('${mandatory + config.bar}, $config.baz')
}

fn main() {
	foo(21,
		bar: 23
		baz: 43
	)
}
```
The following output is produced:
```
44, 43
```

@medvednikov has stated that the short-form syntax should not compile. Source: [Discord](https://discord.com/channels/592103645835821068/700746775962714232/876106635582513213)
Quote:
> `foo(a, bar:b)` shouldn't compile

Note that the code in question is slightly different.

The expected behaviour is to only allow fully qualified struct initialisations. Only when there's one struct present as an argument should the short-form initialisation be allowed.

**Comment by medvednikov** [comment]
There's an open discussion about this.

Good points to allow this syntax are being made.
